queries/flights.py

```diff
@@ -0,0 +1,502 @@
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program. If not, see http://www.gnu.org/licenses/.
+
+
+    This module handles queries relating to air travel.
+
+"""
+
+import re
+import random
+import logging
+import cachetools
+from html import escape
+from typing import List, Dict, Any, Optional
+from datetime import datetime, timedelta, timezone
+
+from query import Query, QueryStateDict
+from queries import query_json_api
+from tree import Result
+from settings import changedlocale
+
+from reynir import NounPhrase
+from geo import capitalize_placename, iceprep_for_placename, icelandic_city_name
+
+
+_FLIGHTS_QTYPE = "Flights"
+
+
+TOPIC_LEMMAS = [
+    "flugvél",
+    "flugvöllur",
+    "flug",
+    "lenda",
+    "brottfarartími",
+    "lendingartími",
+]
+
+
+def help_text(lemma: str) -> str:
+    """Help text to return when query.py is unable to parse a query but
+    one of the above lemmas is found in it"""
+    return "Ég get svarað ef þú spyrð til dæmis: {0}?".format(
+        random.choice(
+            (
+                "Hvenær lendir næsta vél frá Kaupmannahöfn",
+                "Hvenær fer næsta flug til Lundúna",
+            )
+        )
+    )
+
+
+# This module wants to handle parse trees for queries
+HANDLE_TREE = True
+
+# The grammar nonterminals this module wants to handle
+QUERY_NONTERMINALS = {"QFlights"}
+
+# The context-free grammar for the queries recognized by this plug-in module
+GRAMMAR = """
+
+Query →
+    QFlights
+
+QFlights → QFlightsQuery '?'?
+
+QFlightsQuery →
+    QFlightsArrivalQuery
+    | QFlightsDepartureQuery
+
+QFlightsArrivalQuery →
+    QFlightsWhenArr QFlightsNextPlane QFlightsPathDesc
+
+QFlightsWhenArr →
+    QFlightsWhen "lendir"
+    | QFlightsWhen "kemur"
+    | QFlightsWhen "mætir"
+    | "hver" "er" "lendingartími"
+    | "hver" "er" "lendingartíminn" "fyrir"
+
+QFlightsDepartureQuery →
+    QFlightsWhenDepNextPlane QFlightsPathDesc
+
+QFlightsWhenDepNextPlane →
+    QFlightsWhen "leggur" QFlightsNextPlane "af" "stað"
+    > QFlightsWhenDep QFlightsNextPlane "af" "stað"
+    > QFlightsWhenDep QFlightsNextPlane
+
+QFlightsNextPlane →
+    "næsta" QFlightsPlane
+    | "næstu" QFlightsPlane
+
+QFlightsWhenDep →
+    QFlightsWhen "fer"
+    | QFlightsWhen "flýgur"
+    | "hver" "er" "brottfarartími"
+    | "hver" "er" "brottfarartíminn" "fyrir"
+
+$score(+35) QFlightsWhenDep
+
+QFlightsWhen →
+    "hvenær" | "klukkan" "hvað"
+
+$score(-35) QFlightsWhen
+
+QFlightsPlane →
+    "flug" | "flugvél" | "flugvélar" | "vél" | "vélar" | "flugs"
+
+$score(+100) QFlightsPlane
```

Comment:
Not sure that these `$score()` pragmas are needed - is there a reason for them?

Reply:
I removed the score pragmas and it had no effect. I added them to try and fix a bug where queries of the form "Hver er brottfarartími ..." were interpreted as queries asking about a person called "brottfarartími ...". I fixed that elsewhere in the grammar and forgot to remove these :).